### ISSUE TITLE
Set maximum width for the toolbars GUI

### ISSUE THREAD

**Issue by Sebazzz** [OpenLoco/OpenLoco#2428]
**Describe the solution you'd like**
Similar to the reporter of #2249 I'm playing Locomotion on my ultrawide 5120x1440. This works fine *technically*, the game does not crash and there are no weird bugs. However, from an UX perspective it is quite annoying because the toolbar is aligned to the corners of the screen.

Sample screenshot:

![image](https://github.com/OpenLoco/OpenLoco/assets/1426097/7232ee50-8a3b-45f8-9684-763c01019012)

OpenRCT2 has similar problems, but for some reason it is less pronounced. I think they render the GUI perhaps even a bit larger regardless of the render scale, or perhaps it is simply because OpenRCT2 has many more items on the toolbar.

It would be great to have a feature to allow a maximum effective width of the GUI. You see this with other games that support ultrawide that they offer similar settings.

**Describe options you've considered**

- Dropdown with explicit resolutions
- Dropdown with screen ratio's to limit GUI to (for instance 4:3, 6:19)
- Slider so the user can adjust it dynamically and immediately see the result

My preference goes to (3) and should be doable with the existing screen controls.

**Additional context**
Related issue #2249

**Comment by Sebazzz** [comment]
As for #2428: Because windows open in the top left corner they tile nicely when you're opening multiple. I'm thinking that if this would be implemented you'd still constrain the windows to the top left corner, but within the limits set by the user.

**Comment by LeftofZen** [comment]
You could generalise this with #2429 by defining a "UI layer" or "UI bounding box" that is where all the UI controls/windows are placed and rendered by default. This would then totally decouple the UI placement from the background game rendering which should give more flexibility to future features, eg blurring background windows, having filters over the game rendering (but not the UI windows), etc

**Comment by Sebazzz** [comment]
Yes but the bounding box approach might mean you can't drag windows manually outside the bounding box, even if you want it.
